climada/util/api_client.py

```diff
@@ -0,0 +1,472 @@
+        params = {
+            'data_type': data_type,
+            'name': name,
+            'version': version,
+            'status': '' if status is None else status,
+        }
+        params.update(properties if properties else dict())
+        return [DatasetInfo.from_json(ds) for ds in Client._request_200(url, params=params)]
+
+    def get_dataset(self, data_type=None, name=None, version=None, properties=None,
+                    status=None):
+        """Find the one dataset that matches the given parameters.
+
+        Parameters
+        ----------
+        data_type : str, optional
+            data_type of the dataset, e.g., 'litpop' or 'draught'
+        name : str, optional
+            the name of the dataset
+        version : str, optional
+            the version of the dataset
+        properties : dict, optional
+            search parameters for dataset properties, by default None
+        status : str, optional
+            valid values are 'preliminary', 'active', 'expired', and 'test_dataset',
+            by default None
+
+        Returns
+        -------
+        DatasetInfo
+
+        Raises
+        ------
+        AmbiguousResult
+            when there is more than one dataset matching the search parameters
+        NoResult
+            when there is no dataset matching the search parameters
+        """
+        jarr = self.get_datasets(data_type=data_type, name=name, version=version,
+                                 properties=properties, status=status)
+        if len(jarr) > 1:
+            raise Client.AmbiguousResult("there are several datasets meeting the requirements:"
+                                        f" {jarr}")
+        if len(jarr) < 1:
+            raise Client.NoResult("there is no dataset meeting the requirements")
+        return jarr[0]
+
+    def get_dataset_by_uuid(self, uuid):
+        """Returns the data from 'https://climada/rest/dataset/{uuid}' as DatasetInfo object.
+
+        Parameters
+        ----------
+        uuid : str
+            the universal unique identifier of the dataset
+
+        Returns
+        -------
+        DatasetInfo
+
+        Raises
+        ------
+        NoResult
+            if the uuid is not valid
+        """
+        url = f'{self.host}/rest/dataset/{uuid}'
+        return DatasetInfo.from_json(Client._request_200(url))
+
+    def get_data_types(self, data_type_group=None):
+        """Returns all data types from the climada data API
+        belonging to a given data type group.
+
+        Parameters
+        ----------
+        data_type_group : str, optional
+            name of the data type group, by default None
+
+        Returns
+        -------
+        list of DataTypeInfo
+        """
+        url = f'{self.host}/rest/data_types'
+        params = {'data_type_group': data_type_group} \
+            if data_type_group else {}
+        return [DataTypeInfo(**jobj) for jobj in Client._request_200(url, params=params)]
+
```

Comment:
maybe also add a function which allows to get information on all the data_type belonging to a data_type_group, for example get all hazards, or all exposure

Reply:
🤔 ? 
I was thinking that's exactly what `get_data_types` does.

Reply:
Ah yes sorry